scipy/optimize/_differentialevolution.py

```diff
@@ -1231,10 +1241,11 @@ def solve(self):
                 limits[0, integrality] = DE_result.x[integrality]
                 limits[1, integrality] = DE_result.x[integrality]
 
-            polish_method = 'L-BFGS-B'
+            if 'method' not in self.minimizer_kwargs:
+                self.minimizer_kwargs['method'] = 'L-BFGS-B'
 
             if self._wrapped_constraints:
-                polish_method = 'trust-constr'
+                self.minimizer_kwargs['method'] = 'trust-constr'
```

Comment:
This will override a user's desired `method`, if they provide one. Did we want to do that?

Reply:
I changed the code so that the user's preference (if given) overrides the default `trust-constr`